src/tests/Regressions/coreclr/GitHub_86865/test86865.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Reflection;
+
+namespace test86865;
+
+public class test86865
+{
+    public static int Main()
+    {
+
+        // Regression test for https://github.com/dotnet/runtime/issues/86865
+        // Verify that the RuntimeHelpers.GetSpanDataFrom method underlying RuntimeHelpers.CreateSpan<T>
+        // works correctly with enums.
+
+        ReadOnlySpan<MyEnum> myEnums = new[]
+        {
+            MyEnum.A,
+            MyEnum.B,
+            MyEnum.C,
+            MyEnum.B,
+            MyEnum.C,
+        };
+
+        if (string.Join(", ", myEnums.ToArray()) != "A, B, C, B, C")
+            return 1;
+
+        var types = new Type[] {
+            typeof(RuntimeFieldHandle),
+            typeof(RuntimeTypeHandle),
+            typeof(int).MakeByRefType(),
+        };
+        var mi = typeof(System.Runtime.CompilerServices.RuntimeHelpers).GetMethod("GetSpanDataFrom", BindingFlags.Static | BindingFlags.NonPublic, types);
+        if (mi == null)
+            return 2;
+
+        var pid = typeof(MyEnum).Assembly.GetType("<PrivateImplementationDetails>");
+        if (pid == null)
+            return 3;
+
+        var fi = pid.GetField("0B77DC554B4A81403D62BE25FB5404020AD451151D4203D544BF60E3FEDBD8AE", BindingFlags.Static | BindingFlags.NonPublic);
```

Comment:
just curious, is this some kind of internal hash? is it stable?

Reply:
I believe it's a hash of the constant value.  It's fairly stable - in local testing the same field is generated with the net8.0 and net7.0 SDKs when targeting both the net8.0 and net7.0 runtimes.  In CI testing, for some reason the 7.0 branch creates a field with a name that is 1 character longer (ie that fine `"E"` is here on the `release/7.0-staging` version of the test, but not in the `main` branch).

I could just blindly take the first matching field in the assembly (and maybe do some minimal checking of its attributes to make sure it's plausibly the correct one), but it seemed more straightforward to just hardcode the name and to return a distinct failure code and a print out of the available fields.